pkg/acsengine/engine.go

```diff
@@ -431,6 +433,18 @@ func (t *TemplateGenerator) getTemplateFuncMap(properties *api.Properties) map[s
 			str := getBase64CustomScript(swarmWindowsProvision)
 			return fmt.Sprintf("\"customData\": \"%s\"", str)
 		},
+		"GetKubernetesWindowsAgentCustomData": func() string {
+			b, err := Asset(kubernetesWindowsAgentCustomDataPS1)
+			if err != nil {
+				// this should never happen and this is a bug
+				panic(err.Error())
+			}
+			str := string(b)
+			str = strings.Replace(str, "<<<caCertificate>>>", base64.StdEncoding.EncodeToString([]byte(properties.CertificateProfile.CaCertificate)), -1)
```

Comment:
Why? Why are we adding these in customdata when we normally use customscript for this?

Where is the private key?

In general, this is a good example of a place where a new Go template function is really not needed. Whatever we handle the template here should have enough data that it can just splat out the CA Certificate, instead of needing to call a custom Go template function. In some sense... you're basically doing another layer of templating here when you fill in those `<<<placeholders>>>`.